docs/user-guide/commonalities.md

```diff
@@ -0,0 +1,883 @@
+
+- \[5\]: List of volume sources, where volumes refers to the broader term of a
+“volume” as understood by OpenShift.
+
+  This has the exact same structure as the `volumes` section in a `Pod` and is
+  dependent on the type of volume that is being defined, as it is not the same
+  to reference an NFS location as it is to reference a secret.
+
+  The name used here is very important, as it will be used as a reference in the
+  `mounts` section.
+
+  An example of the excerpt of a `volumes` element that defines that the volume
+  is a secret called `ceph-conf-files`:
+
+  ```
+       - name: ceph
+         projected:
+           sources:
+           - secret:
+             name: ceph-conf-files
+  ```
+
+- \[6\]: List of instructions on where the different volumes should be mounted
+within the Pods. Here the name of a volume from the `volumes` section is used as
+a reference as well as the path where it should be mounted.
+
+  This has the exact same structure as the `volumeMounts` in a `Pod`.
+
+  An example of how we would use the `cephf-conf-files` secret that was named as
+  `ceph` in the earlier `volumes` example:
+
+  ```
+       - name: ceph
+         mountPath: "/etc/ceph"
+         readOnly: true
+  ```
+
+Now that we know the `extraMounts` structure and possible values, let’s have a
+look at a couple of examples to tie this all together:
+
+### Example \#1: At the deployment level
+
+A very good example is the propagation of the Ceph configuration that is stored
+in a secret called `ceph-conf-files` to the different storage services (glance,
+manila, and cinder components in the data path):
+
+```
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackControlPlane
+spec:
+  extraMounts:
+    - name: v1
+      region: r1
+      extraVol:
+        - propagation:
+          - CinderVolume
+          - CinderBackup
+          - GlanceAPI
+          - ManilaShare
+          extraVolType: Ceph
+          volumes:
+          - name: ceph
+            projected:
+              sources:
+              - secret:
+                name: ceph-conf-files
+          mounts:
+          - name: ceph
+            mountPath: "/etc/ceph"
+            readOnly: true
+```
+
+### Example \#2: At the component level
+
+Let’s add a custom policy to the cinder API component. We have previously
+created this policy as a `configMap` with the name `my-cinder-policy` and the
+contents are stored in the `policy` key:
+
+```
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackControlPlane
+spec:
+  cinder:
+    extraMounts:
+    - extraVol:
+      - volumes:
+        - name: policy
+          projected:
+            sources:
+            - configMap:
+                name: my-cinder-policy
+                items:
+                - key: policy
+                  path: policy.yaml
+        mounts:
+        - mountPath: /etc/cinder
+          name: policy
+          readOnly: true
+        propagation:
+        - CinderAPI
+```
+
```

Comment:
I think we need to cover the use of `subPath`, which is necessary when the `mountPath` is a directory with other content (not just an empty directory).